modulus/datapipes/gnn/ahmed_body_dataset.py

```diff
@@ -191,83 +208,46 @@ def __init__(
-            graph = self._create_dgl_graph(polydata, outvar_keys, dtype=torch.int32)
-            (
-                velocity,
-                reynolds_number,
-                length,
-                width,
-                height,
-                ground_clearance,
-                slant_angle,
-                fillet_radius,
-            ) = self._read_info_file(info_path)
-            if "velocity" in invar_keys:
-                graph.ndata["velocity"] = velocity * torch.ones_like(
-                    graph.ndata["pos"][:, [0]]
-                )
-            if "reynolds_number" in invar_keys:
-                graph.ndata["reynolds_number"] = reynolds_number * torch.ones_like(
-                    graph.ndata["pos"][:, [0]]
-                )
-            if "length" in invar_keys:
-                graph.ndata["length"] = length * torch.ones_like(
-                    graph.ndata["pos"][:, [0]]
-                )
-            if "width" in invar_keys:
-                graph.ndata["width"] = width * torch.ones_like(
-                    graph.ndata["pos"][:, [0]]
-                )
-            if "height" in invar_keys:
-                graph.ndata["height"] = height * torch.ones_like(
-                    graph.ndata["pos"][:, [0]]
-                )
-            if "ground_clearance" in invar_keys:
-                graph.ndata["ground_clearance"] = ground_clearance * torch.ones_like(
-                    graph.ndata["pos"][:, [0]]
-                )
-            if "slant_angle" in invar_keys:
-                graph.ndata["slant_angle"] = slant_angle * torch.ones_like(
-                    graph.ndata["pos"][:, [0]]
-                )
-            if "fillet_radius" in invar_keys:
-                graph.ndata["fillet_radius"] = fillet_radius * torch.ones_like(
-                    graph.ndata["pos"][:, [0]]
-                )
-
-            if "normals" in invar_keys or self.compute_drag:
-                mesh = pv.read(file_path)
-                mesh.compute_normals(
-                    cell_normals=True, point_normals=False, inplace=True
-                )
-                if "normals" in invar_keys:
-                    graph.ndata["normals"] = torch.from_numpy(
-                        mesh.cell_data_to_point_data()["Normals"]
-                    )
+            self.normals = [None] * self.length
+            self.areas = [None] * self.length
+            self.coeff = [None] * self.length
+
+        # create graphs from VTP files using multiprocessing.
+        def get_num_workers():
+            # Make sure we don't oversubscribe CPUs on a node.
+            # TODO(akamenev): this should be in DistributedManager.
+            local_node_size = max(
+                int(os.environ.get("OMPI_COMM_WORLD_LOCAL_SIZE", 1)), 1
+            )
+            num_workers = len(os.sched_getaffinity(0)) // local_node_size
+            return max(num_workers - 1, 1)
+
+        num_workers = get_num_workers()
```

Comment:
Does it make sense to make this configurable? And then maybe one can use `dist.world_size` for that parameter?

Reply:
Or maybe give an option to overwrite, but if not specified, it uses this implementation. Not sure if I am overthinking this...

Reply:
Good suggestion - made it configurable, with default being auto-select.
Note that `world_size` would not work as expected when the number of nodes > 1 (`world_size` would be the total world size, but we need the local node size), so that's why I have to use `OMPI_COMM_WORLD_LOCAL_SIZE`.